mdn/kuma
# Add subscription component to account settings

Finally, it's here 🤣 

This one turned out to be a lot more involved than I initially thought. Here is the low down.

We used to show your subscription status in two places, after this PR, we will have it in three :/

Keep calm and read further.

It will now be shown on:

1. Edit profile (Jinja based)
2. Payment management page (React)
3. New account settings (React)

For this pull request, there is no need to be concerned with the first, just be aware that it exists. The crux of this pull request, and the majority of the work, is to ensure that versions two and three are essentially a mirror of each other.

I have abstracted and extracted as much as possible to avoid duplication. I have also moved some components from payments to account settings. I did this because once we launch the new account settings page, the payment management page will be completely removed. It therefore, made sense to me to move the relevant components to where they will live post-launch.

Post the new account settings launch we will also completely delete the current edit and view profile pages and as such, will have subscription status and management in one single place o/\o

## Testing

### General code review

While I have pondered over all of this for some time, I have also been staring at it for a long time and there might very well still be an opportunity to optimize this further. I am opening this up now because it has honestly given me "coder's block" :)

### Automated testing

We have good test coverage for all of this so, go ahead and run `yarn test` and ensure all the tests pass.

### Manual testing

Signup for a subscription and then

1. Head over to the payments management page and
1.1 Confirm that the information displayed is correct and matches the design here https://github.com/mdn/kuma/issues/7076#issuecomment-646222210
1.2 Click "Cancel subscription" and confirm that the cancellation form is shown
1.3 Click "Keep subscription" and confirm that the cancellation form is hidden
1.4 Click "Yes, cancel subscription" and confirm that the subscription is successfully canceled

2. Head over to the new account page and
2.1 Follow steps 1.1 - 1.4 above

Also test account settings on mobile, tablet, and desktop viewports.

### Misc

I have also added `@testing-library/user-event` and `@testing-library/dom` because https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-testing-libraryuser-event

### In Closing

Apologies for the big PR :( I cash in 1 of my three large PR chips :)

@Gregoor r?

fix #7076
